pkg/azuredisk/nodeserver.go

```diff
@@ -380,19 +380,54 @@ func (d *Driver) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolume
 	volSizeBytes := int64(capacityBytes)
 	requestGiB := volumehelper.RoundUpGiB(volSizeBytes)
 
-	args := []string{"-o", "source", "--noheadings", "--mountpoint", req.GetVolumePath()}
-	output, err := d.mounter.Exec.Command("findmnt", args...).Output()
+	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "volume path must be provided")
+	}
+
+	// Hopefully devicePath was passed from req
+	devicePath := req.GetStagingTargetPath()
+	if devicePath == "" {
+		args := []string{"-o", "source", "--noheadings", "--mountpoint", volumePath}
+		output, err := d.mounter.Exec.Command("findmnt", args...).Output()
```

Comment:
If the PV.volumemode is block, does this "findmnt" command work?

Reply:
Yes, it will not fail. But the output will not make sense. I think most of the time it just return "" empty. Maybe I should move the isBlock part up